include/irr/asset/filters/dithering/CDither.h

```diff
@@ -47,7 +47,9 @@ namespace irr
 				
 				float get(const IDither::IState* state, const core::vectorSIMDu32& pixelCoord, const int32_t& channel)
 				{
-					return static_cast<CRTP*>(this)->get(static_cast<const typename CRTP::CState*>(state), pixelCoord, channel);
+					const auto& return_value = static_cast<CRTP*>(this)->get(static_cast<const typename CRTP::CState*>(state), pixelCoord, channel);
+					assert(return_value >= 0 && return_value <= 1, "The dither value must be limited to the range!");
```

Comment:
only assert `#ifdef _IRR_DEBUG`

Reply:
also a non-static assert does not take a message argument AFAIK

Reply:
corrected in following commit